pkg/proxy/endpointslicecache_test.go

```diff
@@ -311,6 +311,66 @@ func TestEndpointInfoByServicePort(t *testing.T) {
 				},
 			},
 		},
+		"two slices with duplicate IPs prefer last one to be processed (non local)": {
```

Comment:
I don't think we should say that it actually prefers the last one.... We want to guarantee that we only keep one of them, but it doesn't matter which one.

Things I'd like better than having this test with this name:
- Split this test out into a new function so you can test "when there are duplicate Endpoints, we keep exactly 1" (but the test wouldn't care _which_ one, so that if the code changed later and started keeping the first one instead, the test would still pass).
- Keep the test as-is but rename it to "two serving slices with duplicate IPs, only one is kept", so that if the code changed later and started keeping the first one instead, it would be clearer that it's ok to change the test result to match the new code.
- Remove this test and only keep the "prefer serving" one below

Reply:
we prefer the last one because it will be the more recent received, we can overcomplicate by checking the object timestamps but I think this is just enough , if we have something in the map and we call AddEndpoints, we always overwrite the existing one, and also make the logic deterministic that simplify troubleshooting.

I think the existing patch and test cases cover the bugfix and backports, I prefer to leave for others the refactoring of the code